tools/check_format.py

```diff
@@ -33,6 +33,10 @@
                        './source/common/event/real_time_system.h', './source/exe/main_common.cc',
                        './source/exe/main_common.h', './source/server/config_validation/server.cc',
                        './source/common/common/perf_annotation.h',
+                       './source/extensions/tracers/xray/sampling.h',
```

Comment:
Is this needed? Or can you just work with the TimeSource& provided by the API or other infra?

I think it would be preferable to use the plumbed-in time-source as then you could test with simulated time.

Reply:
I can use plumbed-in time-source in sampling.h file. However, I want to use ` RealTimeSource` in Reservoir implementation. The reason I want to use `RealTimeSource` instead of `TimeSource` there is that it is a counter implementation, I don't want to depend on plumbed-in time-source since that class to be independent and easy to init without providing `time_source_`.

Reply:
I am against adding new elements to this whitelist -- this was designed for the base level infrastructure, and not for new code. Please plumb in time source.